src/core/payments/funding.ts

```diff
@@ -0,0 +1,426 @@
+  depositUSDFC,
+  getPaymentStatus,
+  validatePaymentRequirements,
+  withdrawUSDFC,
+} from './index.js'
+import type {
+  FilecoinPayFundingExecution,
+  FilecoinPayFundingInsights,
+  FilecoinPayFundingPlan,
+  FilecoinPayFundingPlanOptions,
+  FundingMode,
+  FundingReasonCode,
+  PaymentStatus,
+  ServiceApprovalStatus,
+} from './types.js'
+
+function calculateDepletionTiming(
+  available: bigint,
+  perDay: bigint
+): { seconds: bigint; timestampMs?: number | null } | null {
+  if (available <= 0n || perDay <= 0n) {
+    return null
+  }
+
+  const seconds = (available * 86_400n) / perDay
+  if (seconds <= 0n) {
+    return null
+  }
+
+  const maxSecondsForNumber = BigInt(Number.MAX_SAFE_INTEGER) / 1000n
+  const timestampMs = seconds <= maxSecondsForNumber ? Date.now() + Number(seconds) * 1000 : null
+
+  return {
+    seconds,
+    timestampMs,
+  }
+}
+
+/**
+ * Internal helper to build funding insights from payment status
+ */
+function buildFilecoinPayFundingInsights(
+  status: PaymentStatus,
+  overrides?: { depositedBalance?: bigint; rateUsed?: bigint; lockupUsed?: bigint }
+): FilecoinPayFundingInsights {
+  const depositedBalance = overrides?.depositedBalance ?? status.filecoinPayBalance
+  const rateUsed = overrides?.rateUsed ?? status.currentAllowances.rateUsed ?? 0n
+  const lockupUsed = overrides?.lockupUsed ?? status.currentAllowances.lockupUsed ?? 0n
+
+  const runway = calculateStorageRunway({
+    filecoinPayBalance: depositedBalance,
+    currentAllowances: {
+      ...status.currentAllowances,
+      rateUsed,
+      lockupUsed,
+    },
+  })
+
+  const availableDeposited = runway.available
+  const filecoinPayDepletion = calculateDepletionTiming(availableDeposited, runway.perDay)
+  const ownerDepletion = calculateDepletionTiming(availableDeposited + status.walletUsdfcBalance, runway.perDay)
+
+  return {
+    spendRatePerEpoch: rateUsed,
+    spendRatePerDay: runway.perDay,
+    depositedBalance,
+    availableDeposited,
+    walletUsdfcBalance: status.walletUsdfcBalance,
+    runway,
+    filecoinPayDepletionSeconds: filecoinPayDepletion?.seconds ?? null,
+    filecoinPayDepletionTimestampMs: filecoinPayDepletion?.timestampMs ?? null,
+    ownerDepletionSeconds: ownerDepletion?.seconds ?? null,
+    ownerDepletionTimestampMs: ownerDepletion?.timestampMs ?? null,
+  }
+}
+
+/**
+ * Get funding insights for a payment status
+ *
+ * This function calculates runway projections, depletion times, and spend rates
+ * based on current or projected balances and usage.
+ *
+ * @param status - Current payment status
+ * @param overrides - Optional overrides for projected scenarios
+ * @returns Funding insights including runway and depletion predictions
+ */
+export function getFilecoinPayFundingInsights(
+  status: PaymentStatus,
+  overrides?: { depositedBalance?: bigint; rateUsed?: bigint; lockupUsed?: bigint }
+): FilecoinPayFundingInsights {
+  return buildFilecoinPayFundingInsights(status, overrides)
```

Comment:
maybe just delete `buildFilecoinPayFundingInsights` and use `getFilecoinPayFundingInsights` internally

Reply:
obviously I mean inline buildFilecoinPayFundingInsights into getFilecoinPayFundingInsights